### PR TITLE
[Community Pipeline] Add Marigold Monocular Depth Estimation

### PR DESCRIPTION
**Pull request by markkua**
# What does this PR do?

This PR adds a community pipeline of [Marigold](https://marigoldmonodepth.github.io), a universal monocular depth estimator derived from Stable Diffusion.

This pipeline can run on a single GPU and was tested on single RTX 3090/4090 and GTX 1080Ti.

Example code, [project page](https://marigoldmonodepth.github.io), and [full codebase](https://github.com/prs-eth/marigold) are provided. The Colab notebook will be updated to use community pipeline once it is available.

In-the-wild inference result (1st row - input image; 2nd row - predicted depth map; 3rd row - unprojected point cloud):
Marigold Teaser

## Before submitting
- [ ] ~~This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).~~
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] ~~Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.~~
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] ~~Did you write any new necessary tests?~~

## Who can review?

@patrickvonplaten and @sayakpaul

### CODE REVIEW

File: examples/community/README.md [hunk 1]
```diff
@@ -61,6 +62,51 @@ pipe = DiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", custo
...
+
+pipe = DiffusionPipeline.from_pretrained(
+    "Bingxin/Marigold",
+    custom_pipeline="marigold_depth_estimation"
+    # torch_dtype=torch.float16,                # (optional) Run with half-precision (16-bit float).
+)
+
+pipe.to("cuda")
+
+image = Image.open("./input_image.jpg")
```
Comment (sayakpaul) [inline examples/community/README.md:86]: It's better to provide a direct link here and have it loaded like so:

```python
from diffusers.utils import load_image

image = load_image(image_url)
```

**Review by sayakpaul — APPROVED** [review]
Just one comment. Looks very good!

File: examples/community/README.md (hunk 1 above)
Comment (markkua) [reply examples/community/README.md]: Thanks for the suggestion. I've updated the example code with a direct link in README (see the new [commit](https://github.com/huggingface/diffusers/pull/6249/commits/22073d85007dd3742561c9d2241d148b1cf7b4f6)).

**Review by sayakpaul — APPROVED** [review]
Thank you! Will merge after the CI is green.

**Pull request merged**